app/components/UserAvatarAddress.tsx

```diff
@@ -0,0 +1,59 @@
+'use client';
+
+import { shortenAddress } from '@/utils';
+import { cx } from 'class-variance-authority';
+import Image from 'next/image';
+import Link from 'next/link';
+import { useMemo } from 'react';
+import { Address, getAddress } from 'viem';
+import { mainnet } from 'viem/chains';
+import { useEnsAvatar, useEnsName } from 'wagmi';
+
+interface UserAvatarAddressProps {
+  address: Address;
+  isAIAgent: boolean;
+}
+
+export const UserAvatarAddress = ({ address, isAIAgent }: UserAvatarAddressProps) => {
+  const normalizedAddress = useMemo(() => {
+    try {
+      return getAddress(address);
+    } catch (error) {
+      console.error('Invalid Ethereum address:', error);
+      return address;
+    }
+  }, [address]);
+
+  const { data: ensName } = useEnsName({
+    address: normalizedAddress,
+    chainId: mainnet.id,
+  });
+
+  const avatarGradientUrl = (address: string) =>
+    `https://avatars.jakerunzer.com/${address}`;
```

Comment:
What is this URL? Maybe we should give it a name

Reply:
it's the one that give us the gradient based on the address, I'll name it.

Reply:
I'll go with `GRADIENT_BASED_ON_ADDRESS_URL`